Extension/src/LanguageServer/configurations.ts

```diff
@@ -241,26 +243,26 @@ export class CppProperties {
         });
 
         this.configFileWatcher.onDidChange(() => {
-            // If the file is one of the textDocument's vscode is tracking, we need to wait for an
-            // onDidChangeTextDocument event, or we may get old/cached contents when we open it.
-            let alreadyTracking: boolean = false;
-            for (let i: number = 0; i < vscode.workspace.textDocuments.length; i++) {
-                if (vscode.workspace.textDocuments[i].uri.fsPath === settingsPath) {
-                    alreadyTracking = true;
-                    break;
-                }
-            }
-            if (!alreadyTracking) {
-                this.handleConfigurationChange();
-            }
+            this.handleConfigurationChange();
         });
 
-        vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
-            if (e.document.uri.fsPath === settingsPath) {
-                this.handleConfigurationChange();
+        vscode.workspace.onDidChangeTextDocument((e) => {
+            if (e.document.uri.fsPath === settingsPath && this.isCppPropertiesJsonVisible) {
+                this.handleSquiggles();
             }
         });
 
+        vscode.window.onDidChangeVisibleTextEditors((editors) => {
+            wasVisible = this.isCppPropertiesJsonVisible;
```

Comment:
`wasVisible` is only referenced (written, then read) in this lambda function, so it's scope/lifespan can be limited to just this function.  Could you declare the variable here, instead of as a class member?  (Otherwise, the changes look good to me.)